static/app/views/settings/components/forms/selectAsyncField.tsx

```diff
@@ -32,18 +44,24 @@ class SelectAsyncField extends React.Component<SelectAsyncFieldProps> {
     } else if (!Array.isArray(optionObj)) {
       value = optionObj.value;
     }
+    this.setState({latestSelection: optionObj});
     onChange?.(value, event);
     onBlur?.(value, event);
   };
 
-  findValue(propsValue) {
+  findValue(propsValue: string): GeneralSelectValue {
     /**
      * The propsValue is the `id` of the object (user, team, etc), and
      * react-select expects a full value object: {value: "id", label: "name"}
-     *
-     * Returning {} here will show the user a dropdown with "No options".
      **/
-    return this.state.results.find(({value}) => value === propsValue) || {};
+    return (
+      // When rendering the selected value, first look at the API results...
+      this.state.results.find(({value}) => value === propsValue) ??
```

Comment:
This condition would be easier to read if you did 
```tsx
const {defaultOptions} = this.props;
const {latestSelection, results} = this.state;
```
at the top of the function.

Reply:
done!